src/terraform-provider-signalform/signalform/integration.go

```diff
@@ -0,0 +1,116 @@
+package signalform
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"strings"
+)
+
+const (
+	INTEGRATION_API_URL = "https://api.signalfx.com/v2/integration"
+)
+
+func integrationResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the integration",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the integration is enabled or not",
+			},
+			"type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Type of the integration",
+				ValidateFunc: validateIntegrationType,
+			},
+			"api_key": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "PagerDuty API key",
+				Sensitive:   true,
+			},
+			"webhook_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Slack Incoming Webhook URL",
+				Sensitive:   true,
+			},
+		},
+
+		Create: integrationCreate,
+		Read:   integrationRead,
+		Update: integrationUpdate,
+		Delete: integrationDelete,
+	}
+}
+
+func validateIntegrationType(v interface{}, k string) (we []string, errors []error) {
+	value := v.(string)
+	allowedWords := []string{"PagerDuty", "Slack"}
```

Comment:
Would it be a good usability improvement to make this check less strict so also `"pagerduty"` and `Pagerduty` would match the condition?